### PR TITLE
FI_FENCE as user option

### PR DESCRIPTION
**Pull request by wrrobin**
Signed-off-by: Md <md.rahman@intel.com>

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -235,6 +235,13 @@ AC_ARG_ENABLE([memcpy],
     [AC_HELP_STRING([--enable-memcpy],
                     [Use memcpy to perform local put/get operations (default:disabled)])])
 
+AC_ARG_ENABLE([fence],
+    [AC_HELP_STRING([--enable-fence],
+                    [enable and use FI_FENCE instead of transport layer fence in the OFI. (default: disabled)])])
```
Comment (jdinan) [inline configure.ac]: We are starting to use the "ofi" prefix for options tied to the OFI transport, so please use "ofi-fence" for this option. 
 Suggest this wording: "Use FI_FENCE feature to optimize put-with-signal operations. (default: disabled)"

Comment (jdinan) [reply configure.ac]: Looks good apart from this minor nitpick.  👍

**Comment by jdinan** [comment]
Can you also add --enable-ofi-fence to one of the existing testing rows to make sure travis exercises this code?

**Comment by wrrobin** [comment]
@jdinan The reason I did not add any Travis test is that with `FI_FENCE` enabled, the tests will fail because we are requesting for the feature in `fi_getinfo()`. In my local env, it fails for all tests with the following error.
```
[0000] WARN:  transport_ofi.c:1163: query_for_fabric
[0000]        OFI transport did not find any valid fabric services (provider=<auto>)
[0000] ERROR: init.c:383: shmem_internal_init
[0000]        Transport init failed (-61)
[0001] WARN:  transport_ofi.c:1163: query_for_fabric
[0001]        OFI transport did not find any valid fabric services (provider=<auto>)
[0001] ERROR: init.c:383: shmem_internal_init
[0001]        Transport init failed (-61)

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 24200 RUNNING AT node11
=   EXIT CODE: 6
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Aborted (signal 6)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
```
Let me know if there is something missing here.

**Comment by wrrobin** [comment]
Oh.. I just realized for sockets, it should pass... I will check that and make this change.. Ignore my big comment above.

**Comment by wrrobin** [comment]
@jdinan Ready for re-review.

**Pull request merged**